packages/components/Tabs/src/Tabs.tsx

```diff
@@ -0,0 +1,152 @@
+    } = userProps;
+
+    // This hook updates the Selected TabsItem and calls the customer's onTabsClick function. This gets called after a TabsItem is pressed.
+    const data = useSelectedKey(selectedKey || defaultSelectedKey || null, userProps.onTabsClick);
+
+    const [selectedTabsItemRef, setSelectedTabsItemRef] = React.useState(React.useRef<View>(null));
+
+    const onSelectTabsItemRef = React.useCallback(
+      (ref: React.RefObject<View>) => {
+        setSelectedTabsItemRef(ref);
+      },
+      [setSelectedTabsItemRef],
+    );
+
+    const onChangeTabId = React.useCallback((key: string, index: number) => {
+      if (getTabId) {
+        return getTabId(key, index);
+      }
+      return `${key}-Tab${index}`;
+    }, []);
+
+    // Stores views to be displayed
+    const map = new Map<string, React.ReactNode[]>();
+
+    const state: TabsState = {
+      context: {
+        selectedKey: selectedKey ?? data.selectedKey,
+        onTabsClick: data.onKeySelect,
+        getTabId: onChangeTabId,
+        updateSelectedTabsItemRef: onSelectTabsItemRef,
+        views: map,
+      },
+      info: {
+        headersOnly: headersOnly ?? false,
+        label: !!label,
+      },
+    };
+
+    const styleProps = useStyling(userProps, (override: string) => state[override] || userProps[override]);
+
+    const ariaRoles = {
+      accessibilityRole: 'tablist',
+      accessibilityLabel: ariaLabel || label,
+    };
+
+    const slotProps = mergeSettings<TabsSlotProps>(styleProps, {
+      root: { rest, ref: componentRef, ...ariaRoles },
+      label: { children: label },
+      container: { isCircularNavigation: isCircularNavigation, defaultTabbableElement: selectedTabsItemRef },
+    });
+
+    return { slotProps, state };
+  },
+
+  render: (Slots: ISlots<TabsSlotProps>, renderData: TabsRenderData, ...children: React.ReactNode[]) => {
+    if (renderData.state == undefined) {
+      return null;
+    }
+
+    // Populate the tabsItemKeys array
+    if (children) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore - TODO, fix typing error
+      renderData.state.context.tabsItemKeys = React.Children.map(children, (child: React.ReactChild) => {
+        if (React.isValidElement(child)) {
+          // Sets default selected tabItem
+          if (renderData.state.context.selectedKey == null && !child.props.disabled) {
+            renderData.state.context.selectedKey = child.props.itemKey;
+          }
+          return child.props.itemKey;
+        }
+      });
+    }
+
+    return (
+      <TabsContext.Provider
+        // Passes in the selected key and a hook function to update the newly selected tabsItem and call the client's onTabsClick callback
+        value={renderData.state.context}
+      >
+        <Slots.root>
+          {renderData.state.info.label && <Slots.label />}
+          <Slots.container>{children}</Slots.container>
+          <Slots.tabPanel>
+            <TabsContext.Consumer>
```

Comment:
Not blocking: This doesn't seem like a great place to use React's Context - maybe I'm missing something, but to me it looks like all of the information that is provided by the context is immediately here or can be provided to the TabsItem by the Tabs, so I feel like it's a bit overkill to use Context here.

I recognize that you probably followed a pattern from a different part of the codebase, so not blocking since there's precedent for this already.

Reply:
Okay I see that you're keeping children pretty flexible so you can't guarantee that a TabItem is the only child of Tabs. So I guess it makes sense to have a Context to keep things flexible. Though I kind of wonder what's expected to happen if you get a non-TabsItem child

Reply:
If there's a non tabsitem thats a child it will simply just get rendered and it's expected not to affect the functionality of the tabs control. We also expect in general that a user won't try to pass non pivot items.